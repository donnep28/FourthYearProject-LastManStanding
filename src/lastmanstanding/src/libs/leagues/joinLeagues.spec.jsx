import React from 'react';
import {shallow} from 'enzyme';
import JoinLeagues from './joinLeagues';
import TextField from '@material-ui/core/TextField';
import { fireEvent, getByTestId, render } from '@testing-library/react';

describe('joinLeagues', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    user: {},
    setJoinLeague: jest.fn(),
    setRender: jest.fn(),
  };
  describe('Snapshot', () => {
    it('renders teams', () => {
      const component = shallow(
          <JoinLeagues {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('textFields', () => {
    it('updates first name', () => {
      const {getByTestId} = render(<JoinLeagues {...props}/>);
      const element = getByTestId('firstName');
      fireEvent.change(element, {target: {value: 'mockFirstName'}});
      expect(element).toHaveValue('mockFirstName');
    });
    it('updates last name', () => {
      const {getByTestId} = render(<JoinLeagues {...props}/>);
      const element = getByTestId('lastName');
      fireEvent.change(element, {target: {value: 'mockLastName'}});
      expect(element).toHaveValue('mockLastName');
    });
    it('updates league code', () => {
      const {getByTestId} = render(<JoinLeagues {...props}/>);
      const element = getByTestId('leagueCode');
      fireEvent.change(element, {target: {value: 'mockLeagueCode'}});
      expect(element).toHaveValue('mockLeagueCode');
    });
  });
});
