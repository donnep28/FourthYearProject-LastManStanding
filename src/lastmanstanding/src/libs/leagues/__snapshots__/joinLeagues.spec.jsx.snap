// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joinLeagues Snapshot renders teams 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))
    component="main"
    maxWidth="xs"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  >
    <div
      className="makeStyles-paper-1"
    >
      <WithStyles(ForwardRef(Avatar))
        className="makeStyles-avatar-2"
      >
        <Memo(ForwardRef(LockOutlinedIcon)) />
      </WithStyles(ForwardRef(Avatar))>
      <WithStyles(ForwardRef(Typography))
        component="h1"
        variant="h5"
      >
        Join League
      </WithStyles(ForwardRef(Typography))>
      <form
        className="makeStyles-form-3"
        noValidate={true}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={6}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              autoComplete="fname"
              autoFocus={true}
              fullWidth={true}
              id="firstName"
              inputProps={
                Object {
                  "data-testid": "firstName",
                }
              }
              label="First Name"
              name="firstName"
              onChange={[Function]}
              required={true}
              variant="outlined"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={6}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              autoComplete="lname"
              fullWidth={true}
              id="lastName"
              inputProps={
                Object {
                  "data-testid": "lastName",
                }
              }
              label="Last Name"
              name="lastName"
              onChange={[Function]}
              required={true}
              variant="outlined"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              autoComplete="uname"
              fullWidth={true}
              id="leagueName"
              inputProps={
                Object {
                  "data-testid": "leagueCode",
                }
              }
              label="League Code"
              margin="normal"
              name="leagueName"
              onChange={[Function]}
              required={true}
              variant="outlined"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-submit-4"
          color="primary"
          fullWidth={true}
          onClick={[Function]}
          type="submit"
          variant="contained"
        >
          Join League
        </WithStyles(ForwardRef(Button))>
      </form>
    </div>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
